### PR TITLE
fix(parser): reject a repeated bare node pattern in CREATE

### LINKED ISSUE #2274: `CREATE (n),(n)` silently creates one node instead of detecting duplicate variable declaration
**FalkorDB version**

4.20.1

**Environment**

- OS: Windows with Docker Desktop
- Architecture: x86_64
- Deployment: Docker container

**Describe the bug**

FalkorDB shows inconsistent behavior when handling duplicate variable declarations inside a single `CREATE` clause.

In Neo4j and Memgraph, reusing the same variable name within the same query scope is rejected because the variable has already been declared.

For example, the following query is rejected:

```cypher
CREATE (n),(n)
````

Neo4j and Memgraph report an error similar to:

```
Variable `n` already declared
```

However, FalkorDB accepts the same query and creates only one node:

```cypher
CREATE (n),(n)
```

Result:

```
Nodes created: 1
```

This behavior is inconsistent with both Neo4j/Memgraph and with FalkorDB's own handling of similar cases.

For comparison, FalkorDB correctly detects duplicate variable declarations when they appear in separate `CREATE` clauses:

```cypher
CREATE (n) CREATE(n)
```

Result:

```
(error) The bound variable 'n' can't be redeclared in a CREATE clause
```

While:

```cypher
CREATE (n) CREATE(m)
```

correctly creates two nodes:

```
Nodes created: 2
```

This indicates that duplicate variable validation works in one code path (`CREATE` clauses separated by keywords), but is missing or inconsistent when multiple patterns are provided in the same `CREATE` clause using commas.

**Expected behavior**

`CREATE (n),(n)` should be rejected because the variable `n` is redeclared within the same query scope.

The expected behavior should be consistent with Neo4j and Memgraph:

```
Error: variable `n` already declared
```

It should have deterministic behavior consistent with Cypher semantics. It should not silently create a different number of nodes.

### PR DESCRIPTION
**Pull request by olifarhaan**
## TLDR

`CREATE (n),(n)` created one node and reported success. A pattern keeps one node per alias, so the second `(n)` was dropped on the way in and the redeclaration was never visible to any check

Neo4j and Memgraph reject it as a redeclaration, and FalkorDB already rejects the same mistake written other ways: `CREATE (n) CREATE (n)` and the relationship form `CREATE ()-[e:R]->()-[e:R]->()` both error with `The bound variable ... can't be redeclared in a CREATE clause`. The inconsistency was confined to this one shape

This reports it the same way, off the value `add_pattern_node` already computes about whether the alias is new to the pattern

Reusing an alias to attach a relationship is a reference rather than a declaration, so `CREATE (a),(b),(a)-[:R]->(b)` stays valid. The check therefore only fires on a path that opens on a bare repeat, decided by peeking for the relationship token the path parser is about to read anyway

Unaliased nodes need no exemption. Each is given a fresh `_anon_<n>` from a per-parse counter, so its alias is unique and it passes the "alias is new" test on its own. An earlier revision exempted the `_anon` prefix by name, which left `CREATE (_anon),(_anon)` creating one node silently, since a query may write that name itself

## User Flow

`GRAPH.QUERY` with a `CREATE` that names the same variable twice as a standalone node pattern

## Relevant issues

Closes #2274

## Pre-Submission checklist

- [x] I have added meaningful tests
- [x] My PR passes all CI/CD checks (e.g., lint, format, unit tests)
- [x] My PR's scope is as isolated as possible; it only solves 1 specific problem

## Screenshots / Proof of Fix

Live server, release build, module loaded into a real `redis-server`

```bash
redis-server --port 6399 --loadmodule "$PWD/target/release/libfalkordb.dylib" &
for q in "CREATE (n),(n)" "CREATE p = (n), q = (n)" "CREATE (n)-[:R]->(m), (n)" \
         "CREATE (a),(b),(a)-[:R]->(b)" "CREATE (a)-[:R]->(b), (b)-[:S]->(c)" "CREATE (),()"; do
  redis-cli -p 6399 GRAPH.DELETE c > /dev/null 2>&1
  echo "\$ $q"; redis-cli -p 6399 GRAPH.QUERY c "$q" | head -3; echo
done
```

Before, on this branch's merge base:

```text
$ CREATE (n),(n)
Nodes created: 1

$ CREATE p = (n), q = (n)
Nodes created: 1

$ CREATE (n)-[:R]->(m), (n)
Nodes created: 2
Relationships created: 1

$ CREATE (_anon),(_anon)
Nodes created: 1
```

After:

```text
$ CREATE (n),(n)
The bound variable 'n' can't be redeclared in a CREATE clause

$ CREATE p = (n), q = (n)
The bound variable 'n' can't be redeclared in a CREATE clause

$ CREATE (n)-[:R]->(m), (n)
The bound variable 'n' can't be redeclared in a CREATE clause

$ CREATE (a),(b),(a)-[:R]->(b)
Nodes created: 2
Relationships created: 1

$ CREATE (a)-[:R]->(b), (b)-[:S]->(c)
Nodes created: 3
Relationships created: 2

$ CREATE (),()
Nodes created: 2

$ CREATE (_anon),(_anon)
The bound variable '_anon' can't be redeclared in a CREATE clause

$ CREATE (),(),()
Nodes created: 3
```

## Type

🐛 Bug Fix

## Changes

`graph/src/parser/cypher.rs`, `add_pattern_node` now returns whether the alias was new, and a `reject_create_redeclaration` helper is applied at the two sites that parse the node a path opens on

`tests/flow/test_create_clause.py`, one regression test covering the rejected shapes, the alias a query could collide with, and the still-valid ones

## QA runbook

The new test fails on the merge base and passes on this branch, with the three pre-existing tests in the file unaffected either way

```text
# merge base
Total Tests Run: 4, Total Tests Failed: 1, Total Tests Passed: 3
	Unhandled exception: 'CREATE (n),(n)' should have been rejected, created 1.0 nodes

# this branch
Total Tests Run: 4, Total Tests Failed: 0, Total Tests Passed: 4
```

The test raises rather than calling `self.env.assertTrue(False)`, which is the idiom `test02_edge_reuse` above it uses for the same purpose. That idiom does not reliably fail a test, so a regression test built on it can report PASS while its assertion is firing

Full flow suite: `Total Tests Run: 1409, Total Tests Failed: 23, Total Tests Passed: 1386`, identical before and after. All 23 fail inside redis-py 8.1.0 with `Redis.__init__() got an unexpected keyword argument 'himport_registry'`, a client-side version mismatch in my local venv that cannot be reached from the module

TCK passes. `tests/test_e2e.py`, `tests/test_functions.py` and `tests/test_overflow_crashes.py`: 142 passed. `cargo test -p graph`: 116 passed. `cargo fmt --all --check` clean, `CXX=clang++ cargo clippy --all-targets` exits 0 leaving only the pre-existing `too many arguments` warning in `cond_var_len_traverse.rs`

One judgement call worth a look in review. `CREATE (n)-[:R]->(m), (n)` is now rejected, since the trailing `(n)` opens a path on an alias the clause already declared. Nothing is lost by it, that trailing pattern could never have created a second node, but if the preference is to accept a bare repeat once the alias participates in a relationship anywhere in the clause, that is a one-line change to the same helper

### Final Attestation

- [x] The tests check the right things, including the edge cases, and regressions in the respective real-world customer use-cases are not possible after this PR

## Summary by CodeRabbit

* **Bug Fixes**
  * `CREATE` clauses now reject repeated declarations of the same node alias.
  * Existing aliases can still be reused as relationship endpoints.
  * Anonymous nodes remain supported while invalid patterns are prevented.

* **Tests**
  * Added coverage for alias reuse, redeclaration errors, and expected node and relationship counts.

### CODE REVIEW

File: graph/src/parser/cypher.rs [hunk 1]
```diff
@@ -1042,17 +1042,48 @@ impl<'a> Parser<'a> {
...
+            || *clause != Keyword::Create
+            || node.alias.starts_with("_anon")
+            || matches!(self.lexer.current()?, Token::Dash | Token::LessThan)
+        {
+            return Ok(());
+        }
+        Err(format!(
+            "The bound variable '{}' can't be redeclared in a CREATE clause",
+            node.alias
+        ))
```
[the remark this answers is not in the document]
Comment (olifarhaan) [reply graph/src/parser/cypher.rs:1090]: Right, fixed in 8e0dccb. Dropped the prefix exemption entirely rather than tracking anonymity: generated aliases are already unique, so it was dead code.

[the remark this answers is not in the document]
Comment (olifarhaan) [reply graph/src/parser/cypher.rs:1090]: Also narrowed the test to catch ResponseError only, in 89e2ec5. Catching Exception could have masked the failure the test exists to catch.
